src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderChapter.jsx

```diff
@@ -0,0 +1,373 @@
+  let yesNoKey;
+  for (const key of Object.keys(uiSchema)) {
+    if (uiSchema[key]['ui:webComponentField'] === YesNoField) {
+      yesNoKey = key;
+    }
+  }
+  return yesNoKey;
+}
+
+/**
+ * arrayBuilderChapter
+ *
+ * Example:
+ * ```
+ * chapterName: arrayBuilderChapter(pageBuilder => {
+ *   title: 'Employment history',
+ *   options: {
+ *     arrayPath: 'employers',
+ *     noun: 'employer',
+ *     nextChapterPath: '/next-chapter',
+ *   }
+ *   summaryPage: pageBuilder.summaryPage({
+ *     title: 'Employment history',
+ *     path: '/summary',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ *   itemFirstPage: pageBuilder.itemFirstPage({
+ *     title: 'Name and address of employer or unit',
+ *     path: '/employer',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ *   itemLastPage: pageBuilder.itemLastPage({
+ *     title: 'Name and address of employer or unit',
+ *     path: '/employer',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ * }
+ * ```
+ *
+ * @param {(pageBuilder: ArrayBuilderPages) => ArrayBuilderConfig} configFunction
+ * @returns {FormConfigChapter}
+ */
+export default function arrayBuilderChapter(configFunction) {
+  let itemFirstPagePath;
+  let summaryPath;
+  let hasItemsKey;
+
+  if (!configFunction || typeof configFunction !== 'function') {
+    throwErrorNoConfigFunction();
+  }
+
+  /**
+   * @type {{
+   *   [key: string]: (config: FormConfigPage) => FormConfigPage,
+   * }}
+   */
+  const pageBuilder = {
+    summaryPage: pageConfig => {
+      summaryPath = pageConfig.path;
+      try {
+        hasItemsKey = determineYesNoField(pageConfig.uiSchema);
+      } catch (e) {
+        throwMissingYesNoField();
+      }
+      if (!hasItemsKey) {
+        throwMissingYesNoField();
+      }
+    },
+    itemFirstPage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+      itemFirstPagePath = getBasePath(pageConfig.path);
+    },
+    itemMiddlePage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+    itemLastPage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+    itemSinglePage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+  };
+
+  // run configFunction once to get initial options
+  const { options } = configFunction(pageBuilder);
+  if (!options) {
+    throwErrorNoOptions();
+  }
+
+  const requiredChapterOpts = ['arrayPath', 'noun', 'nextChapterPath'];
+  verifyRequiredOptions('chapter', requiredChapterOpts, options);
+
+  const {
+    noun,
+    arrayPath,
+    nextChapterPath,
+    reviewPath = '/review-and-submit',
+    isIncompleteItem = item => !item?.name, // TODO: update this to use user param
+  } = options;
+
+  const nounSingular = noun;
+  const nounPlural = pluralize(noun);
```

Comment:
I feel like it might be easier to include both of these as options instead of using a `pluralize` function, e.g. `noun: ['employer', 'employers']`. 

There are _a lot_ of [irregular plurals](https://github.com/plurals/pluralize/blob/master/pluralize.js#L260) and words with no plurals. Also, I'm not sure how we'd handle this once the VA adds translations.

Reply:
hm, maybe I'll just stick to nounSingular/nounPlural for now (had this in the previous commit)